### PR TITLE
rbd-mirror: remove volume replication sidecar

### PR DESCRIPTION
**Pull request by yati1998**
The volume replication operator is being moved to
kubernetes-csi-addons. This commit hence, removes
the volume replication sidecar and updates the
related documentation.

Fixes: #10655

Signed-off-by: yati1998 <ypadia@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
Does this need to wait till we get the new csi-addons release?

File: deploy/examples/common.yaml
Comment (Madhu-1) [inline deploy/examples/common.yaml:163]: RBACs are still required isn't it?

**Comment by yati1998** [comment]
> Does this need to wait till we get the new csi-addons release?

Yes, we should have one. 
cc @Rakshith-R @nixpanic

File: deploy/examples/common.yaml
Comment (yati1998) [reply deploy/examples/common.yaml:163]: yes, we need it. Forgot to revert the change. Will update it.

File: Documentation/Storage-Configuration/Block-Storage-RBD/rbd-mirroring.md
Comment (Madhu-1) [inline Documentation/Storage-Configuration/Block-Storage-RBD/rbd-mirroring.md:169]:
```
kubectl create -f https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/main/config/crd/bases/replication.storage.openshift.io_volumereplicationsclasses.yaml
error: unable to read URL "https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/main/config/crd/bases/replication.storage.openshift.io_volumereplicationsclasses.yaml", server reported 404 Not Found, status code=404
```

the correct URL is `kubectl create -f https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/main/config/crd/bases/replication.storage.openshift.io_volumereplicationclasses.yaml`

File: deploy/examples/common.yaml
Comment (Madhu-1) [reply deploy/examples/common.yaml:163]: Looks like RBAC is required only for the controller, and controller RBACs are installed separately we can remove this one but need testing and confirmation

File: Documentation/Storage-Configuration/Block-Storage-RBD/rbd-mirroring.md
Comment (travisn) [inline Documentation/Storage-Configuration/Block-Storage-RBD/rbd-mirroring.md:169]: We will have a versioned link soon, right? Referencing main would be less stable.

**Review by travisn — CHANGES_REQUESTED** [review]
- See the CI errors
- Will any upgrade steps be necessary to transition from the existing replication to the new solution? As long as the CRD is not being renamed, I assume no special steps are needed.

File: Documentation/Storage-Configuration/Block-Storage-RBD/rbd-mirroring.md
Comment (yati1998) [reply Documentation/Storage-Configuration/Block-Storage-RBD/rbd-mirroring.md:169]: yes, once we have csi-addons release I will update them too.

**Comment by yati1998** [comment]
> * See the CI errors
Yeah, will check them.

> * Will any upgrade steps be necessary to transition from the existing replication to the new solution? As long as the CRD is not being renamed, I assume no special steps are needed.
I don't think so we require any upgrade steps. The name of the crds is gonna be the same.

**Comment by yati1998** [comment]
@nixpanic @Rakshith-R can you please review the PR? Will update the csi-addons with the versioned URL, once the build is completed.

**Review by subhamkrai — CHANGES_REQUESTED** [review]
please change this https://github.com/rook/rook/blob/6610aa8699ad5b9bd1a6df433ba14d2f3ed14c1f/tests/framework/installer/ceph_installer.go#L153 too check golangci lint error

**Review by Rakshith-R — CHANGES_REQUESTED** [review]
please update k8s-sidecar image version too.

File: Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md
Comment (travisn) [inline Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md:175]: These URLs use main, but the other URLs on above lines have v0.3.0. Is there a `latest` tag (or similar) instead? At least we should make them consistent.

Comment (travisn) [reply Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md:175]: Above I see some other links in this PR to v0.5.0. Is that the latest version?

File: tests/integration/ceph_smoke_test.go
Comment (travisn) [inline tests/integration/ceph_smoke_test.go:96]: Could we enable the CSI addons in the tests instead of completely removing this option? That might be a separate PR though.

**Review by travisn — CHANGES_REQUESTED** [review]
Are the

File: Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md
Comment (yati1998) [reply Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md:175]: oh yeah, we have the latest version v0.5.0 out, forgot to update it for reclaimspace and network fence. Will do that. Thanks @travisn for mentioning it.

**Review by Rakshith-R — CHANGES_REQUESTED** [review]
> please update k8s-sidecar image version too.

This is still not addressed

File: Documentation/Helm-Charts/operator-chart.md
Comment (Rakshith-R) [inline Documentation/Helm-Charts/operator-chart.md:44]: Too many lines have changed, 

try editing with normal text editor, to preserve other lines.

**Review by Rakshith-R — CHANGES_REQUESTED** [review]
ci runs are failing,
almost looks good to me, left one comment

**Review by travisn — CHANGES_REQUESTED** [review]
For the commit prefix, use `rbdmirror` instead of `rbd-mirror`, this changed recently in #10821.

File: Documentation/Helm-Charts/operator-chart.md
Comment (Madhu-1) [reply Documentation/Helm-Charts/operator-chart.md:44]: please fix this one. @yati1998

File: Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md
Comment (Madhu-1) [inline Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md:88]: You also need to mention user needs to create a few more csiaddons CRD and install the csi-addons controller?

File: tests/framework/installer/ceph_installer.go
Comment (Madhu-1) [inline tests/framework/installer/ceph_installer.go:157]: not requried?

File: Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md
Comment (yati1998) [reply Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md:88]: yeah. maybe we can redirect to the later part of the documentation rather than repeating it.

Comment (Madhu-1) [reply Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md:88]: sounds good

**Comment by Madhu-1** [comment]
@yati1998 you need to remove volume replication RBAC, i don't see that change anymore in this PR

**Comment by yati1998** [comment]
> @yati1998 you need to remove volume replication RBAC, i don't see that change anymore in this PR

Removing the RBAC changes was resulting in ci failure, hence reverted that. I will test removing the RBAC changes and if required will send another PR for it.

**Review by Madhu-1 — COMMENTED** [review]
changes LGTM, please remove volume replication RBAC.

File: Documentation/Storage-Configuration/Block-Storage-RBD/rbd-mirroring.md
Comment (nixpanic) [inline Documentation/Storage-Configuration/Block-Storage-RBD/rbd-mirroring.md:191]: CSI-Addons is the name of the project, maybe use that like you do later in this document?

File: Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md
Comment (nixpanic) [inline Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md:82]: `CSI-Addons` is the correct spelling, also needs fixing on the next line

**Review by nixpanic — COMMENTED** [review]
Looks good to me, for CSI-Addons changes. Just a few nits regarding the docs.

File: Documentation/Storage-Configuration/Block-Storage-RBD/rbd-mirroring.md
Comment (yati1998) [reply Documentation/Storage-Configuration/Block-Storage-RBD/rbd-mirroring.md:191]: @nixpanic how about having a separate cleanup PR for it? For now, we are following this everywhere else in the repo. Changing at one place would require changes else where too.

**Comment by subhamkrai** [comment]
@yati1998 I was looking to the failure I see this in the logs
```
ceph-csi: failed to reconcile failed to configure ceph csi: failed to start ceph csi drivers: failed to load rbd provisioner deployment template: failed to load deployment template: template: rbd-provisioner:124:14: executing "rbd-provisioner" at <.EnableVolumeReplicationSideCar>: can't evaluate field EnableVolumeReplicationSideCar in type csi.templateParam
```
there is no csi pods created

**Review by Madhu-1 — APPROVED** [review]

**Review by travisn — APPROVED** [review]

**Review by subhamkrai — APPROVED** [review]
LGTM after merge conflicts resolved @yati1998

**Comment by Madhu-1** [comment]
Remove https://github.com/rook/rook/blob/master/build/crds/build-crds.sh#L54-L67 and https://github.com/rook/rook/blob/master/build/crds/build-crds.sh#L130-L132 as we dont need it anymore?

**Review by Rakshith-R — APPROVED** [review]
Thanks!

**Review by Madhu-1 — APPROVED** [review]

File: pkg/operator/ceph/csi/spec.go
Comment (Rakshith-R) [inline pkg/operator/ceph/csi/spec.go:134]: This should be v0.5.0

**Review by Rakshith-R — CHANGES_REQUESTED** [review]
@yati1998
 
ci is failing 
Please run make again 
the requested change should fix it too.

**Review by Rakshith-R — APPROVED** [review]

**Pull request merged**
